lib/hybrid_platforms_conductor/hpc_plugins/platform_handler/serverless_chef/recipes_tree_builder.rb

```diff
@@ -0,0 +1,271 @@
+require 'hybrid_platforms_conductor/hpc_plugins/platform_handler/serverless_chef/dsl_parser'
+
+module HybridPlatformsConductor
+
+  module HpcPlugins
+
+    module PlatformHandler
+
+      class ServerlessChef < HybridPlatformsConductor::PlatformHandler
+
+        # Build the recipes tree from a ServerlessChef platform
+        class RecipesTreeBuilder
+
+          # Constructor
+          #
+          # Parameters::
+          # * *config* (Config): Configuration that can be used to tune tree building
+          # * *platform* (ServerlessChef): Platform for which we build the recipes tree
+          def initialize(config, platform)
+            @config = config
+            @platform = platform
+          end
+
+          # Get the whole tree of recipes
+          #
+          # Result::
+          # * The tree of recipes:
+          #   Hash< Symbol,   Hash< Symbol, Hash<Symbol,Object> >
+          #   Hash< cookbook, Hash< recipe, recipe_info         >
+          #   Each recipe info has the following attributes:
+          #   * *included_recipes* (Array< [String or nil, Symbol, Symbol] >): List of [cookbook_dir, cookbook, recipe] included by this recipe
+          #   * *used_by_policies* (Array<String>): List of policies that include (recursively) this recipe
+          #   * *used_templates* (Array<String>): List of template sources used by this recipe
+          #   * *used_files* (Array<String>): List of cookbook files used by this recipe
+          #   * *used_cookbooks* (Array<Symbol>): List of additional used cookbooks (for example for resources)
+          def full_recipes_tree
+            @recipes_tree = {}
+            @platform.deployable_services.each do |service|
+              policy_run_list(service).each do |(cookbook_dir, cookbook, recipe)|
+                add_recipe_in_tree(cookbook_dir, cookbook, recipe)
+              end
+            end
+            @platform.deployable_services.each do |service|
+              policy_run_list(service).each do |(cookbook_dir, cookbook, recipe)|
+                mark_recipe_used_by_policy(cookbook, recipe, service)
+              end
+            end
+            @recipes_tree
+          end
+
+          private
+
+          # Get the run list of a given policy
+          #
+          # Parameters::
+          # * *policy* (String): Policy to get the run list from
+          # Result::
+          # * Array<[String or nil, Symbol,Symbol]>: Run list of the given policy, as [cookbook_dir, cookbook, recipe]
+          def policy_run_list(policy)
```

Comment:
Might be covered by knife deps --tree.

Reply:
Tried to use it in local mode but couldn't get a meaningful output yet. Might have to investigate this deeper.